File: Frontend/src/services/Student_Service.js
import axios from "axios";
import Auth_Header from './Auth_Header';

const BASE_URL = "http://localhost:8080/students/";


class Student_Service {
    getAllStudents(pageNumber, pageSize, sortBy, sortDir) {
        return axios.get(BASE_URL + "?pageNumber=" + pageNumber + "&pageSize=" + pageSize + "&sortBy=" + sortBy + "&sortDir=" + sortDir, { headers: Auth_Header() })
    }
    delete(id) {
        return axios.delete(BASE_URL + id + "/", { headers: Auth_Header() })
    }
    getStudentbyId(id) {
        return axios.get(BASE_URL + id + "/", {
            headers: Auth_Header()
        })
    }
    count() {
        return axios.get(BASE_URL + "count", {
            headers: Auth_Header()
        })
    }
    editStudent(id, student) {
        return axios.put(BASE_URL + id + "/", student, {
            headers: Auth_Header()
        })
    }
}
export default new Student_Service;

File: Backend/src/main/java/com/example/demo/services/StudentService.java
package com.example.demo.services;

import java.util.List;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

import com.example.demo.models.StudentModel;

public interface StudentService {
	
	//View all students
	public ResponseEntity<List<StudentModel>> viewAllStudents(Integer pageNumber,Integer pageSize,String sortBy,String sortDir);
	
	//View admission by id
	public ResponseEntity<StudentModel> viewAdmissionById(Long studentid);
	
	//View student by studentId
	public ResponseEntity<List<StudentModel>> viewStudentById(Long studentid);
	
	//Count of all students
	public Long getCount();
	
	//Edit student
	public ResponseEntity<List<StudentModel>> editStudentById(Long studentid , StudentModel student);
	
	
	//Delete student
	public ResponseEntity<HttpStatus> deleteStudentModel(Long studentid);
}
